### ISSUE TITLE
help does not work

### ISSUE THREAD

**Issue by andreygrozin** [wxMaxima-developers/wxmaxima#1981]
I'm on Gentoo linux (I'm the maintainer of the wxmaxima package in this distro). Recently I've installed 24.11.0 configured with -DWXM_DISABLE_WEBVIEW=ON (without this, an old version of webkit-gtk is required, namely 4.1; another program on my computer wants the modern version 6; in principle, it is possible to have both of them installed at the same time, but each version compiles a few hours, and compiling both is too tiresome).

1. I click Help -> Maxima help. My browser is started, and says that it cannot open
`file:/usr/share/maxima/5.47.0/doc/html/maxima_singlepage.html`
This file is, in fact, at
`/usr/share/doc/maxima-5.47.0-r1/html/maxima_singlepage.html`
I remember that some time ago there was a workaround in the code to find html help files at their Gentoo locations, and it used to work; now it does not work.

2. I click Help -> wxMaxima help. A dialog box appears saying
> Allow to access a online manual for maxima?
Didn't find an installed offline manual.

In fact, the wxMaxima manual is installed at
`/usr/share/doc/wxmaxima-24.11.0/wxmaxima.html`
(a couple of minor grumblings about this dialog: (1) why _maxima_, not _wxMaxima_? (2) _a online manual_ should be _an online manual_). OK, I allow to access an online manual, and the browser cannot open
`https://htmlpreview.github.io/?https://github.com/wxMaxima_developers/wxmaxima/blob/main/info/wxmaxima.html`

In other words, help does not work in this version.

**Comment by daute** [comment]
Hi,
concerning (1): Where does the path (especially the `-r1`) come from? Is your Maxima configured so 'special' (it seems the Maxima ./configure option `--htmldir` has no effect. I already reported this some time ago: https://sourceforge.net/p/maxima/bugs/3696/)? Or was that renamed later?

Concerning (2): That probably affects many more programs, not only wxMaxima. It seems, that browsers (e.g. Chrome) are blocked to access local files (e.g. something in `/usr/share`). See e.g.: https://github.com/netblue30/firejail/issues/5125
Great. Access to local installed files (which **everyone** on the system can access according to the Unix file permissions) are blocked. But every crap from the internet is happily downloaded and accessed.
Must be a great security improvement. However - I do **not** understand it.

Concerning `https://htmlpreview.gitsrc/wxMaxima.cpphub.io/?https://github.com/wxMaxima_developers/wxmaxima/blob/main/info/wxmaxima.html` - that should be a `-` not an underscore in the URL. I really wonder, where that comes from, in the source we have the `-`. (https://github.com/wxMaxima-developers/wxmaxima/blob/Version-23.11.0/src/wxMaxima.cpp#L4690). You are using an unpatched wxMaxima-sourcecode?

**Comment by andreygrozin** [comment]
`-r1` means revision 1. `maxima` installed on my notebook is `5.47.0-r1`. This means that the upstream version is `5.47.0`, and the revision number of the Gentoo package based on this upstream version is `r1`. This is similar to what, say, Debian is doing: there the current maxima is, I think, `5.47.0-11`, i.e. the upstream version `5.47.0`, the revision of the Debian package based on it is 11. The documentation of this package in Gentoo is in `/usr/share/doc/maxima-5.47.0-r1`. I.e., the file we want is `/usr/share/doc/maxima-5.47.0-r1/html/maxima_singlepage.html`.
`wxmaxima` installed on my notebook is 24.11.0. It is revision 0, and -r0 is never written anywhere (this is the most frequent situation). I've committed it to Gentoo yesterday. So, the file we want is `/usr/share/doc/wxmaxima-24.11.0/wxmaxima.html`.
So, the question is: is it possible to explain to wxmaxima to search the help files at `/usr/share/doc/maxima-<version><maybe -r*>/html/maxima_singlepage.html` and `/usr/share/doc/wxmaxima-<version><maybe -r*>/wxmaxima.html`.
The remaining problems. I've tried to instruct the system to use `google-chrome` for displaying html files. I did `xdg-mime default google-chrome.desktop text/html`. I added the files called `mimeapps.list` to both `/etc/xdg/` and `/usr/share/applications/` with the line `text/html=google-chrome.desktop;`.  `xdg-open foo.html` opens `foo.html` in `google-chrome`. OK, this is what I want. But wxmaxima help system starts `crqt` instead of `google-chrome`. `crqt` is a viewer of electronic books, and yes, it can display html. `crqt` is the alphabetically first program in `/usr/share/applications/` which contains `text/html` in the `MimeType` field of its `.desktop` file. All `xdg` things are completely ignored. `crqt` is not a browser, and it does not understand an url as a parameter, just a plain file name. Is it possible to convince `wxmaxima` to use `google-chrome` for displaying html, not something different? Ideally, to follow the `xdg` configuration?
Sorry, the wrong url of `wxmaxima.html` at `github` was my mistake. I've rewritten it incorrectly.

**Comment by daute** [comment]
Hi,
Concerning the help browser: Edit->Configure->Maxima ->Help browser.
Select "User specified" and then `xdg-open`, Chrome, Firefox or whatever browser you prefer.
The default from the wxWidgets-Framework (Autodetect) is maybe to choose the alphabetical first possibility.

Concerning the path to the help files: One possibility would be to make these paths configurable too. The paths you expect are not really that one, which are the default after a Maxima or wxMaxima installation. Maybe fix that with symlinks?

Or - to work around obscure bugs, that browsers can not open lowebkit-gtkcal (readable) files - an option to always use the online version?
Does `google-chrome file:///usr/share/doc/wxmaxima-24.11.0/wxmaxima.html` really work?

Best regards & a happy new year, Wolfgang

**Comment by andreygrozin** [comment]
Thank you very much! Setting the help browser to `xdg-open` solves the problem.
I've found that the maxima package actually installs the documentation to two places: `/usr/share/doc/maxima--5.47.0-r1` and `/usr/share/maxima/5.47.0/doc`. And Help -> Maxima help successfully displays `maxima_singlepage.html` from the second directory. I'm also the maintainer of the `maxima` package; I should improve it by replacing one of these two directories by the simlink to the other one.
`xdg-open /usr/share/doc/wxmaxima-24.11.0/wxmaxima.html` works fine.
Help -> wxMaxima help successfully displays the html from github, because some time ago I allowed it to get it from the web. So, I cannot test if it works locally, because I don't know how to revoke this permission.
Maybe, it would be good to make the directories of the help files configurable?

**Comment by daute** [comment]
Concerning how to revoke the network help permission: In the wxMaxima configuration file (`~/.wxMaxima` or `.config/wxMaxima.conf` (depending on the wxWidgets version) there is an entry:
`allowNetworkHelp=1`
Change that to zero.
